aospy/utils/times.py

```diff
@@ -571,3 +503,89 @@ def ensure_time_as_index(ds):
             da[TIME_STR] = ds[TIME_STR]
             ds[name] = da
     return ds
+
+
+def infer_year(date):
+    """Given a datetime-like object or string infer the year
+
+    Parameters
+    ----------
+    date : str
+        Input date
+
+    Returns
+    -------
+    int
+    """
+    if isinstance(date, str):
+        pattern = '(?P<year>\d{4})'
+        result = re.match(pattern, date)
+        if result:
+            return int(result.groupdict()['year'])
+        else:
+            raise ValueError('Invalid date string provided: {}'.format(date))
+    elif isinstance(date, np.datetime64):
+        return date.item().year
+    else:
+        return date.year
+    
+
+def maybe_cast_as_timestamp(date):
+    """Convert a date to a pd.Timestamp object if possible
+
+    Parameters
+    ----------
+    date : datetime-like object
+        Input datetime
+
+    Returns
+    -------
+    pd.Timestamp or cftime.datetime
```

Comment:
Describe the conditions under which either type is returned.

Reply:
Thanks for catching this; this function was was leftover from my earlier commits and is no longer even used!  I'll go ahead and delete it.